src/Model/Scope/Condition.php

```diff
@@ -0,0 +1,384 @@
+        $value = $condition->replaceValue($condition->value);
+
+        if (!$condition->isActive()) {
+            return [];
+        }
+
+        if ($model = $condition->model) {
+            if (is_string($field)) {
+                // shorthand for adding conditions on references
+                // use chained reference names separated by "/"
+                if (stripos($field, '/') !== false) {
+                    $references = explode('/', $field);
+
+                    $field = array_pop($references);
+
+                    foreach ($references as $link) {
+                        $model = $model->refLink($link);
+                    }
+
+                    // '#' -> has # referenced records
+                    // '?' -> has any referenced records
+                    // '!' -> does not have any referenced records
+                    if (in_array($field, ['#', '!', '?'], true)) {
+                        // if no operator consider '#' as 'any record exists'
+                        if ($field == '#' && !$operator) {
+                            $field = '?';
+                        }
+
+                        if (in_array($field, ['!', '?'], true)) {
+                            $operator = '=';
+                            $value = $field == '?' ? 1 : 0;
+                        }
+                    } else {
+                        // otherwise add the condition to the referenced model
+                        // and check if any records exist matching the criteria
+                        $model->addCondition($field, $operator, $value);
+                        $operator = '=';
+                        $value = 1;
+                    }
+
+                    // if not counting we check for existence only
+                    $field = $field == '#' ? $model->action('count') : $model->action('exists');
+                } else {
+                    $field = $model->getField($field);
+                }
+            }
+
+            // @todo: value is array
+            // convert the value using the typecasting of persistence
+            if ($field instanceof Field && $model->persistence && !in_array(strtoupper((string) $operator), self::$skipValueTypecast, true)) {
+                $value = $model->persistence->typecastSaveField($field, $value);
+            }
+
+            // only expression contained in $field
+            if (!$operator) {
+                return [$field];
+            }
+
+            // skip explicitly using '=' as in some cases it is transformed to 'in'
+            // for instance in dsql so let exact operator be handled by Persistence
+            if ($operator === '=') {
+                return [$field, $value];
+            }
+        }
+
+        return [$field, $operator, $value];
+    }
+
+    public function isEmpty(): bool
+    {
+        return array_filter([$this->key, $this->operator, $this->value]) ? false : true;
+    }
+
+    public function validate(array $values): array
```

Comment:
very inconsistent with anythink else - Model->set is validate immediatelly

this should always validate, even if disabled

Reply:
It cannot as scope is stand-alone and may not be associated with model to validate to

Reply:
I can imagine this situation, but once it is associated...  Also, it is active by default, so it will fail when not associated by default...